### DISCUSSION TITLE
Stripping Exif

### DISCUSSION THREAD

**Discussion by sethify** [libvips/libvips#2326] [Q&A] [answered]
Hello!

We are working on a web server that transforms images. Sometimes there is no work to be done and we pass the image through. We need to strip exif data (quickly) when there is no work to do. 

I found this: https://github.com/libvips/libvips/issues/1026

The example above implies that the source image is output, untouched except for the exif data.

We are trying to do this without disk files, as in we use only buffers. We've tried `vips_image_write_to_memory ()` but found we are receiving a very large buffer (is this raw image data?). We also looked at `vips_image_write_to_buffer ()` but this function seems to only output a few types and requires a suffix making it different than `vips_image_write_to_file ()` .

Is there a buffer based equivalent to `vips_image_write_to_file()`? 

Thank you.

**Comment by sethify** [comment]
I should add, that we actually want to preserve some exif data, such as ICC profile,  `page-height`, `n-pages`, `loop`, `delay` and orientation.

**Comment by jcupitt** [comment, marked as answer]
Hi @sethify,

libvips will always decode and recode images, so a specialist exif editor might be a better fit.

But you can avoid a huge memory intermediate. In Python, for example:

```python
im = pyvips.Image.new_from_buffer(source, "", access ="sequential")
im = im.copy()
for field in im.get_fields():
    if field not in whitelist:
        im.remove(field)
dest = im.write_to_buffer(".jpg")
```

Will copy the jpg from `source` to `dest` removing every metadata item not in `whitelist` and only need a small amount of memory.

**Reply by sethify** [reply]
I was afraid that might be the answer. For our use case we don't want to go through the lossy process of recoding the image just to ensure there are no privacy leaks. I've been poking around at the file formats and this doesn't look very fun.

**Reply by sethify** [reply]
Speed is our variable of concern, memory isn't really an issue because our instances become processor bound before we see any memory pressure (thanks to the great work you have already done).
